O/OpenCL/build_tarballs.jl

```diff
@@ -0,0 +1,51 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "OpenCL"
+version = v"2022.09.23"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/KhronosGroup/OpenCL-ICD-Loader.git", "3dae4803532b11d74e4dc216ee72570c1a4bff24")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+
+install_license ./OpenCL-ICD-Loader/LICENSE
+
+cmake -DCMAKE_PREFIX_PATH=${prefix} -DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -S ./OpenCL-ICD-Loader -B ./OpenCL-ICD-Loader/build
+cmake --build ./OpenCL-ICD-Loader/build --target install -j${nproc}
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl")
+]
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libOpenCL", :libcl)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("OpenCL_Headers_jll", v"2022.09.23")
```

Comment:
I [asked](https://github.com/JuliaPackaging/Yggdrasil/pull/5607#discussion_r984803681) for this to be a [`BuildDependency`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.BuildDependency)
```suggestion
    BuildDependency(PackageSpec(; name="OpenCL_Headers_jll", version=v"2022.09.23")),
```